File: library/src/main/java/com/thesurix/gesturerecycler/GestureListener.java
package com.thesurix.gesturerecycler;


import android.support.v7.widget.helper.ItemTouchHelper;

/**
 * Default gesture listener that handles manual spawned drag gestures.
 * @author thesurix
 */
public class GestureListener implements GestureAdapter.OnGestureListener {

    private final ItemTouchHelper mTouchHelper;

    public GestureListener(final ItemTouchHelper touchHelper) {
        mTouchHelper = touchHelper;
    }

    @Override
    public void onStartDrag(final GestureViewHolder viewHolder) {
        mTouchHelper.startDrag(viewHolder);
    }
}

File: library/src/main/java/com/thesurix/gesturerecycler/GestureListener.kt
package com.thesurix.gesturerecycler


import androidx.recyclerview.widget.ItemTouchHelper

/**
 * Default gesture listener that handles manual spawned drag gestures.
 * @author thesurix
 */
class GestureListener(private val touchHelper: ItemTouchHelper) : GestureAdapter.OnGestureListener<Any> {

    override fun onStartDrag(viewHolder: GestureViewHolder<Any>) {
        touchHelper.startDrag(viewHolder)
    }
}
